src/promise.js

```diff
@@ -35,3 +35,36 @@ export function pause (seconds) {
     setTimeout(resolve, seconds * 1000)
   })
 }
+
+/**
+ * Polls for the presence of a value fetched by a function, within a deadline.
+ * @param {Object} params
+ * @param {number} params.interval Interval between fetchValue checks, in secondsgit
+ * @param {number} params.deadline Deadline, in seconds
+ * @param {function} fetchValue
+ * @returns {Promise}
+ */
+export async function poll ({ interval, deadline }, fetchValue) {
+  let deadlineElapsed = false
+  let handle = null
+  if (deadline) {
+    handle = window.setTimeout(() => { deadlineElapsed = true }, deadline * 1000)
+  }
+
+  const cleanup = () => {
+    if (handle) {
+      window.clearTimeout(handle)
+    }
+  }
+
+  while (!deadlineElapsed) {
+    const lastValue = fetchValue()
+    if (lastValue) {
```

Comment:
we should check `lastValue != undefined`. The poll should be able to stop on falsy values